src/NServiceBus.AcceptanceTesting/ContextAppender.cs

```diff
@@ -7,150 +7,130 @@
-            });
-        }
-
-        static void AppendException(Exception exception)
-        {
-            context?.LoggedExceptions.Enqueue(exception);
+            contextGetter()?.LoggedExceptions.Enqueue(exception);
         }
 
         public void Debug(string message)
         {
-            //we don't care about debug logs
+            Log(message, LogLevel.Debug);
         }
 
         public void Debug(string message, Exception exception)
         {
             AppendException(exception);
+            Log(message, LogLevel.Debug);
         }
 
         public void DebugFormat(string format, params object[] args)
         {
-            //we don't care about debug logs
+            var fullMessage = string.Format(format, args);
+            Log(fullMessage, LogLevel.Debug);
         }
 
         public void Info(string message)
         {
-            Trace.WriteLine(message);
-            RecordLog(message, "info");
+            Log(message, LogLevel.Info);
         }
 
 
         public void Info(string message, Exception exception)
         {
             var fullMessage = $"{message} {exception}";
-            Trace.WriteLine(fullMessage);
+            Log(fullMessage, LogLevel.Info);
             AppendException(exception);
-            RecordLog(fullMessage, "info");
         }
 
         public void InfoFormat(string format, params object[] args)
         {
             var fullMessage = string.Format(format, args);
-            Trace.WriteLine(fullMessage);
-            RecordLog(fullMessage, "info");
+            Log(fullMessage, LogLevel.Info);
         }
 
         public void Warn(string message)
         {
-            Trace.WriteLine(message);
-            RecordLog(message, "warn");
+            Log(message, LogLevel.Warn);
         }
 
         public void Warn(string message, Exception exception)
         {
             var fullMessage = $"{message} {exception}";
-            Trace.WriteLine(fullMessage);
+            Log(fullMessage, LogLevel.Warn);
             AppendException(exception);
-            RecordLog(fullMessage, "warn");
         }
 
         public void WarnFormat(string format, params object[] args)
         {
             var fullMessage = string.Format(format, args);
-            Trace.WriteLine(fullMessage);
-            RecordLog(fullMessage, "warn");
+            Log(fullMessage, LogLevel.Warn);
         }
 
         public void Error(string message)
         {
-            Trace.WriteLine(message);
-
-            RecordLog(message, "error");
+            Log(message, LogLevel.Error);
         }
 
         public void Error(string message, Exception exception)
         {
             var fullMessage = $"{message} {exception}";
-            Trace.WriteLine(fullMessage);
+            Log(fullMessage, LogLevel.Error);
             AppendException(exception);
-            RecordLog(fullMessage, "error");
         }
 
         public void ErrorFormat(string format, params object[] args)
```

Comment:
not really related to this PR, but I know we have code which does pass the exception on the args instead of using the exception parameter (if there are other parameters to log together with the exception). Technically we could probably check whether args contains an exception and append it to the exceptions too. Not sure whether that's a good idea though.